orc/src/main/java/org/apache/iceberg/orc/OrcIterable.java

```diff
@@ -41,29 +41,31 @@
   private final InputFile file;
   private final Long start;
   private final Long length;
+  private final OrcRowFilter rowFilter;
   private final Function<TypeDescription, OrcValueReader<?>> readerFunction;
 
   OrcIterable(InputFile file, Configuration config, Schema schema,
               Long start, Long length,
-              Function<TypeDescription, OrcValueReader<?>> readerFunction) {
+              Function<TypeDescription, OrcValueReader<?>> readerFunction, OrcRowFilter rowFilter) {
     this.schema = schema;
     this.readerFunction = readerFunction;
     this.file = file;
     this.start = start;
     this.length = length;
     this.config = config;
+    this.rowFilter = rowFilter;
   }
 
   @SuppressWarnings("unchecked")
   @Override
   public Iterator<T> iterator() {
     Reader orcFileReader = ORC.newFileReader(file, config);
     addCloseable(orcFileReader);
+    addCloseable(rowFilter);
```

Comment:
why should a filter be closable, isn't it stateless

Reply:
Actually our row filter is based on the bitmap, which needs to be cleaned up when the task finishes.